karate-core/src/test/java/com/intuit/karate/core/DummyUiTest.java

```diff
@@ -0,0 +1,49 @@
+package com.intuit.karate.core;
+
+import com.intuit.karate.Runner;
+import com.intuit.karate.Suite;
+import com.intuit.karate.TestUtils;
+import com.intuit.karate.core.runner.NoopDriver;
+import com.intuit.karate.driver.DriverRunner;
+import com.intuit.karate.http.HttpClient;
+import com.intuit.karate.http.HttpRequest;
+import com.intuit.karate.http.Response;
+import com.intuit.karate.report.Report;
+import com.intuit.karate.report.SuiteReports;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+/**
+ *
+ * @author pthomas3
+ */
+public class DummyUiTest {
+
+    FeatureRuntime fr;
+
+    public static FeatureRuntime runFeature(String path) {
+        Map<String, DriverRunner> customDrivers = new HashMap<>();
+        customDrivers.put(NoopDriver.DRIVER_TYPE, NoopDriver::start);
+        Feature feature = Feature.read(path);
+        Runner.Builder rb = Runner.builder();
+        rb.features(feature);
+        rb.configDir("classpath:com/intuit/karate/core");
+        rb.customDrivers(customDrivers);
```

Comment:
are you missing a change ? I don't see this in the driver. or can we just limit this PR to the logging change and nothing else. it is a bit confusing

Reply:
@ptrthomas the unit test covers https://github.com/karatelabs/karate/issues/2009#issuecomment-1295897586 . Just piggybacked on this branch that I had open, sorry for confusion.

The Noop driver is something I had introduced a while ago to interact with the driver functions without actually opening a UI so we could test in a pipeline the interaction with those methods. Think it's useful for testing the framework access to methods and behaviors by "mocking" expected responses from browsers (i.e. in theory we could unit test the devtools driver this way, if we wanted)

Reply:
updated description of this PR for future traceability

Reply:
cool thanks, honestly hadn't noticed `builder.customDrivers()` etc. makes sense, and all good